File: laminatoria-client/src/app/classes/product.ts
import { Category } from './category'
import { Properties } from './properties'

export class Product {
	id: number
	name: string
	vendor: string
	category: Category
	properties: Properties[]
	additionalProperty: AdditionalProperty //запрашивать с сервера
	typeOfProduct: number
	typeOfMeasurement: number
	price: number
	relatedProductsId: number[]
	images: string

	constructor(
		id: number,
		name: string,
		vendor: string,
		category: Category, //ламинат, линолеум и тд
		properties: Properties[],
		additionalProperty: AdditionalProperty,
		typeOfProduct: number, // отрезной или штучный
		typeOfMeasurement: number, // руб.шт руб.м2 руб.м
		price: number,
		images: string
	) {
		this.id = id
		this.name = name
		this.vendor = vendor
		this.category = category
		this.properties = properties
		this.additionalProperty = additionalProperty
		this.typeOfProduct = typeOfProduct
		this.typeOfMeasurement = typeOfMeasurement
		this.price = price
		this.images = images
	}
}

export class AdditionalProperty {
	id: number
	name: string
	values: string

	constructor(id: number, name: string, values: string) {
		this.id = id
		this.name = name
		this.values = values
	}
}

export enum typeOfProduct {
	'cutting',
	'units',
}

export const typeOfProductMap: Map<number, string> = new Map()

typeOfProductMap.set(0, 'Отрезной')
typeOfProductMap.set(1, 'Штучный')

export enum typeOfMeasurement {
	'roublesForSquareMeter',
	'roublesForUnit',
}

export const typeOfMeasurementMap: Map<number, string> = new Map()

typeOfMeasurementMap.set(0, 'р./м2')
typeOfMeasurementMap.set(1, 'р./шт.')

File: Laminatoria/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Laminatoria.Models
{
    public class Product
    {
        [Key]
        public int Id { get; set; }
        public string Name { get; set; }
        public string Vendor { get; set; }
        public TypeOfMeasurement TypeOfMeasurement { get; set; }
        public TypeOfProduct TypeOfProduct { get; set; }
        public short? CategoryId { get; set; }
        public Category? Category { get; set; }        
        public AdditionalProperty AdditionalProperty { get; set; }

        [Column(TypeName = "decimal(9,2)")]
        public decimal Price { get; set; }
        public string Images { get; set; }
        public List<Properties> Properties { get; set; } = new List<Properties>();
    }

    public enum TypeOfMeasurement
    {
        RoublesForSquareMeter,
        RoublesForUnit
    }

    public enum TypeOfProduct
    {
        cutting,
        units
    }
}
